packages/core/scripts/setup/patches/common.ts

```diff
@@ -0,0 +1,85 @@
+import EmbraceLogger from '../../../src/logger';
+import { FileUpdatable } from '../../util/file';
+import { OBJECTIVEC_APP_DELEGATE } from './ios/ios.objectivec';
+import { SWIFT_APP_DELEGATE } from './ios/ios.swift';
+
+// TODO refactor this
+const logger = new EmbraceLogger(console);
+export const JAVA_MAIN_ACTIVITY = 'MainApplication.java';
+export const KOTLIN_MAIN_ACTIVITY = 'MainApplication.kt';
+
+/**
+ * Adds a line to the specified file after the specified search text.
+ *
+ * This function will return true if the line was successfully added to the file.
+ *
+ * @param file The file where the new line will be added.
+ * @param lineToAdd The line that will be added to the file.
+ * @param searchText The text used to determine the location where the new line will be added.
+ *                   The new line will be added after the first occurrence of this text.
+ * @param fileName The name of the file, used only for logging purposes.
+ * @return Returns true if the line was successfully added to the file, false if the line already exists.
+ */
+const addLineAfterToTextInFile = (
+  file: FileUpdatable,
+  lineToAdd: string,
+  searchText: string | RegExp,
+  fileName?: string
+): boolean => {
+  if (file.hasLine(lineToAdd)) {
+    logger.warn(`${fileName} already contains the line: ${lineToAdd}`);
+    return false;
+  } else {
+    logger.warn(`Adding: ${lineToAdd} in ${fileName}`);
+    file.addAfter(searchText, lineToAdd);
+    return true;
+  }
+};
+
+/**
+ * Adds a line to the specified file after the specified search text.
+ *
+ * This function will return true if the line was successfully added to the file.
+ *
+ * @param file The file where the new line will be added.
+ * @param lineToAdd The line that will be added to the file.
+ * @param searchText The text used to determine the location where the new line will be added.
+ *                   The new line will be added after the first occurrence of this text.
+ * @param fileName The name of the file, used only for logging purposes.
+ * @return Returns true if the line was successfully added to the file, false if the line already exists.
+ */
+const addLineBeforeToTextInFile = (
+  file: FileUpdatable,
+  lineToAdd: string,
+  searchText: string | RegExp,
+  fileName?: string
+): boolean => {
+  if (file.hasLine(lineToAdd)) {
+    logger.warn(`${fileName} already contains the line: ${lineToAdd}`);
+    return false;
+  } else {
+    logger.warn(`Adding the line: ${lineToAdd} in ${fileName}`);
+    file.addBefore(searchText, lineToAdd);
```

Comment:
might be simpler to have a single function and take 'before'|'after' as a parameter

Reply:
Following single responsibility principle it should be 2 function. But I can change it if you want